tools/dochack/dochack.nim

```diff
@@ -2,16 +2,17 @@ import dom
 import fuzzysearch
 
 
-proc switchTheme(event: Event) =
-  if event.target.checked:
-    document.documentElement.setAttribute("data-theme", "dark")
-    window.localStorage.setItem("theme", "dark")
-  else:
-    document.documentElement.setAttribute("data-theme", "light")
-    window.localStorage.setItem("theme", "light")
+proc setTheme(theme: cstring) {.exportC.} =
+  document.documentElement.setAttribute("data-theme", theme)
```

Comment:
Is it recommended to use `cstring` explicitly ?

Reply:
I think so.

Reply:
cstring".." instead of ".." everywhere ? this seems inconvenient. maybe there should be a function with a shorter name like ^"..".

Reply:
String literals are implicitly converted to cstring when possible. You don't always have to explicitly write `cstring`.

If you were asking if this is recommended or not, I would say it is.

Reply:
The manual says:
> Even though the conversion is implicit, it is not _safe_: The garbage collector does not consider a cstring to be a root and may collect the underlying memory. For this reason, the implicit conversion will be removed in future releases of the Nim compiler. Certain idioms like conversion of a const string to cstring are safe and will remain to be allowed.

So since it will be removed we need to write cstring".." everywhere. maybe there need to be a shorter way to write a cstring literal.

Reply:
> Certain idioms like conversion of a const string to cstring are safe and will remain to be allowed.

Reply:
the lines just above:

> A Nim string is implicitly convertible to cstring for convenience. If a Nim string is passed to a C-style variadic proc, it is implicitly converted to cstring too:
> ```nim
> proc printf(formatstr: cstring) {.importc: "printf", varargs,
>                                  header: "<stdio.h>".}
>
> printf("This works %s", "as expected")
> ```

The same usage is discouraged isn't it ?

Reply:
nvm I got it.
> If you were asking if this is recommended or not, I would say it is.

Why, since it'll remain to be allowed ?

Reply:
Because of the exact reason you want to use it. It's less boilerplate, and more convenient to call procs like this that work with the backend/optimize by using cstring. It's like, if there is a `proc foo(x: uint)`, assuming no other overloads, should you be forced to call it as `foo(1.uint)` or should you be allowed to do `foo(1)`?

Reply:
What I understood from your sentence is that using `cstring` explicitly is recommended even though it's more boilerplate. that's why I asked for why you think it's recommended.